rpi/board.go

```diff
@@ -69,8 +69,15 @@ func (conf *Config) Validate(path string) ([]string, error) {
 			return nil, err
 		}
 	}
-	for idx, c := range conf.DigitalInterrupts {
-		if err := c.Validate(fmt.Sprintf("%s.%s.%d", path, "digital_interrupts", idx)); err != nil {
+
+	for _, c := range conf.Pins {
+		if c.Pin == "" {
+			return nil, resource.NewConfigValidationFieldRequiredError(path, "pin")
+		}
+		if c.Name == "" {
+			return nil, resource.NewConfigValidationFieldRequiredError(path, "name")
```

Comment:
I actually think name should be optional, because you can use the `DigitalInterruptByName` or `GPIOPinByName` right now without having ascribed a name for the pin, and access it with its number.